cvat/apps/engine/tests/test_lazy_list.py

```diff
@@ -170,14 +165,31 @@ def test_length_on_iteration(self):
         self.assertEqual(elements, [1, 2, 3])
 
     def test_str(self):
-        self.assertEqual(str(self.lazy_list), "1,2,3")
+        self.assertEqual(str(self.lazy_list), "1,2,,3")
         self.assertEqual(self.lazy_list, LazyList(str(self.lazy_list), converter=int))
 
     def test_str_parsed(self):
         list(self.lazy_list)
         self.assertEqual(str(self.lazy_list), "1,2,3")
         self.assertEqual(self.lazy_list, LazyList(str(self.lazy_list), converter=int))
 
+    def test_slice(self):
+        self.assertEqual(self.lazy_list[1:], [2, 3])
+        self.assertEqual(self.lazy_list[:2], [1, 2])
+        self.assertEqual(self.lazy_list[::2], [1, 3])
+        self.assertEqual(self.lazy_list[1::2], [2])
+        self.assertEqual(self.lazy_list[::-1], [3, 2, 1])
+        self.assertEqual(self.lazy_list[-2:], [2, 3])
+        self.assertEqual(self.lazy_list[:-2], [1])
+        self.assertEqual(self.lazy_list[:-1], [1, 2])
+        self.assertEqual(self.lazy_list[-2::-1], [2, 1])
+
+    def test_slice_with_stride(self):
+        self.assertEqual(self.lazy_list[::2], [1, 3])
+        self.assertEqual(self.lazy_list[::-2], [3, 1])
+        self.assertEqual(self.lazy_list[1::-1], [2, 1])
+        self.assertEqual(self.lazy_list[1:3:2], [2])
+
```

Comment:
It looks like it's better to split these test cases further or merge them together. I'd also added a more long list and an empty one.

Reply:
Good point. I've unified this in one test case, but splitted each one in a separate subtest.

And made list longer for all test cases.